_source/_posts/2020-12-09-dotnet-cloud-host-publish.md

```diff
@@ -0,0 +1,325 @@
+[Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+*AWS deployment:*
+[AWS account](https://aws.amazon.com)
+
+*Google Cloud deployment:*
+[Google Cloud](https://cloud.google.com) account
+[Google Cloud SDK](https://cloud.google.com/sdk)
+
+## Preparing the .NET Core App for Cloud Host deployment
+
+I have prepared an example application to play around with and see how the deployments work.
+
+Start by cloning the repo to your local machine:
+
+```sh
+git clone https://github.com/oktadeveloper/okta-netcore3-deploy-to-cloud-hosts-example
+```
+
+Since there is Okta Authentication in the example application, I first need to update the related configs.
+
+Log in to your [Okta Developer account](https://login.okta.com)
+
+Navigate to **Applications**, then select **Add Application**.
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image1.png alt:"" width:"500" %}{: .center-image }
+
+Select **Web** as a platform:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image2.png alt:"" width:"800" %}{: .center-image }
+
+On the next screen leave everything as is and select Done.
+
+Scroll down and note your Okta Client Credentials:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image3.png alt:"" width:"700" %}{: .center-image }
+
+Open *appsettings.json* in your favorite code editor and add your credentials.
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image4.png alt:"" width:"500" %}{: .center-image }
+
+You can find your Org URL in the top right corner of the Dashboard:
+
+{% img blog/dotnetcore-deploy-major-cloud-hosts/image5.png alt:"" width:"800" %}{: .center-image }
+
+Now your app should be ready for deployment and to run locally. Let's make sure there are no build or runtime errors. Open a terminal or command prompt, go to the project folder and run:
+
+```sh
+dotnet run
+```
+
+Open your browser at `https://localhost:5001` to see the app running.
+
+>*The Login part of the app doesn't work just now. We will set it up after the deployments.*
+
+## Deploying a .NET Core 3.1 App to Azure App Service
+
+Let's begin by deploying our app to Azure!
+
+### Creating an Azure App Service using Command Line
+
+Open a terminal or command prompt and login to Azure.
+
+```sh
+az login
+```
+
+A browser window will open asking for your Microsoft Account credentials.
+
+First, we'll have to add a Resource Group:
+
+```sh
+az group create --name myResourceGroup --location westus
+```
+
+Run the following command to set the default resource group and location for all subsequent commands. So we won't have to worry about adding them at each command.
+
+```sh
+az configure --defaults group=myResourceGroup location=westus
+```
+
+Let's create an App Service plan:
+
+```sh
+az appservice plan create --name myPlan --sku F1
+```
+
+Finally, create the App Service itself:
+
+Replace <your_app_name> with your preferred name. This will be the URL of your app, like: `http://<your_app_name>.azurewebsites.net`
```

Comment:
It looks like you might need raw tags - or don't use angle brackets since they disappear. From the preview.

![image](https://user-images.githubusercontent.com/17892/101671564-e1146f80-3a11-11eb-8fce-7ee4ed679781.png)